packages/status-js/src/client/activityCenter.ts

```diff
@@ -0,0 +1,143 @@
+// todo?: rename to notifications (center?), inbox, or keep same as other platforms
+
+import type { ChatMessage } from './chat'
+import type { Client } from './client'
+
+// todo?: rename to Activity
+type Notification = {
+  type: 'message'
+  value: ChatMessage
+  isMention?: boolean
+  isReply?: boolean
+}
+
+export type ActivityCenterLatest = {
+  notifications: Notification[]
+  // todo?: rename count to mentionsAndRepliesCount
+  unreadChats: Map<string, { count: number }>
+}
+
+export class ActivityCenter {
+  #client: Client
+
+  #notifications: Set<Notification>
+  #callbacks: Set<(latest: ActivityCenterLatest) => void>
+
+  constructor(client: Client) {
+    this.#client = client
+
+    this.#notifications = new Set()
+    this.#callbacks = new Set()
+  }
+
+  public getLatest = (): ActivityCenterLatest => {
+    const notifications: Notification[] = []
+    const unreadChats: Map<string, { count: number }> = new Map()
+
+    for (const notification of this.#notifications.values()) {
+      if (notification.type === 'message') {
+        const chatUuid = notification.value.chatUuid
+
+        const chat = unreadChats.get(chatUuid)
+        let count = chat?.count ?? 0
+
+        if (notification.isMention || notification.isReply) {
+          count++
+        }
+
+        if (chat) {
+          chat.count = count
+        } else {
+          unreadChats.set(chatUuid, { count })
+        }
+      }
+
+      notifications.push(notification)
+    }
+
+    notifications.sort((a, b) => {
+      if (a.value.clock > b.value.clock) {
+        return -1
+      }
+
+      if (a.value.clock < b.value.clock) {
+        return 1
+      }
+
+      return 0
+    })
+
+    // fixme!?: do not display regular messages, only mentions and replies
+    // todo?: group notifications (all, unreads, mentions, replies, _chats.{id,count})
+    return { notifications, unreadChats }
+  }
+
+  // todo: pass ids instead of values and resolve within
```

Comment:
tried as poc and reverted at https://github.com/status-im/status-web/pull/303/commits/c9f33a70c4f74a9ab0c1307ac926639e08410275